reviews/api/serializers.py

```diff
@@ -0,0 +1,100 @@
+from proposals.models import Proposal
+from uil.core.rest.serializers import ModelDisplaySerializer
+from ..models import Decision, Review
+from rest_framework import serializers
+from main.serializers import UserSerializer
+
+
+class ReviewProposalSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Proposal
+        fields = ['pk', 'reference_number', 'title', 'is_revision',
+                  'date_confirmed', 'date_submitted', 'parent', 'latest_review',
+                  'applicants']
+
+    parent = serializers.SerializerMethodField()
+    latest_review = serializers.SerializerMethodField()
+    applicants = serializers.SerializerMethodField()
+
+    def get_parent(self, proposal):
+        if proposal.parent:
+            return ReviewProposalSerializer(proposal.parent).data
+
+        return None
+
+    def get_latest_review(self, proposal):
+        review = proposal.latest_review()
+
+        if review:
+            return InlineReviewSerializer(review).data
+
+        return None
+
+    def get_applicants(self, proposal):
+        return UserSerializer(proposal.applicants.all(), many=True).data
+
+
+class InlineDecisionSerializer(ModelDisplaySerializer):
+    class Meta:
+        model = Decision
+        fields = ['pk', 'go', 'date_decision', 'comments', 'reviewer']
+
+    reviewer = serializers.SerializerMethodField()
+
+    def get_reviewer(self, decision):
+        return UserSerializer(decision.reviewer).data
+
+
+class InlineReviewSerializer(ModelDisplaySerializer):
+    class Meta:
+        model = Review
+        fields = ['pk', 'stage', 'route', 'go', 'continuation', 'date_start',
+                  'date_end', 'date_should_end', 'accountable_user',
+                  'current_reviewers']
+
+    route = serializers.CharField(source='get_route_display')
+    accountable_user = serializers.SerializerMethodField()
+    current_reviewers = serializers.SerializerMethodField()
+
+    def get_accountable_user(self, review):
+        return UserSerializer(review.accountable_user()).data
+
+    def get_current_reviewers(self, review):
+        return [user.pk for user in review.current_reviewers()]
+
+
```

Comment:
What's the difference between an inline and regular serializer?

Reply:
You mean the difference between for example ``InlineDecisionSerializer`` and ``DecisionSerializer``?

The difference between them is the fact that the inline versions don't have references to their counterpart Review (or counterpart Decision in case of ``InlineReviewSerializer`` and the connected proposal.

This is needed, as using the regular ``DecisionSerializer`` in ``ReviewSerializer`` would generate an infinite recursion. (As the decision serializer would in turn include a review, which would then include the decision set again, which would then include the review, etc). 
The proposal isn't included for a similar reason, we don't need multiple copies of a proposal. 

Thus, in short, inline is a subset of the regular version such that it can be included in a different regular serializer.